REPOSITORY-GUIDELINES.md

```diff
@@ -1,101 +1,194 @@
+
+It is important to the success of Knative that all Knative repositories stay
+active, healthy, and aligned with the scope and mission of project.
+
+### Grounds for removal
+
+Core repositories may be removed from the project if they are deemed _inactive_.
+Inactive repositories are those that meet any of the following criteria:
+
+- There are no longer any active maintainers for the project, and no
+  replacements can be found.
+- All PRs and issues have gone un-addressed for longer than six months.
+- There have been no new commits or other changes in more than a year.
+- The contents have been folded into another actively maintained project.
+
+### Procedure for removal
+
+When a repository has been deemed eligible for removal, we take the following
+steps:
+
+- A proposal to remove the repository is brought to the attention of the
+  Technical Oversight Committee through a GitHub issue posted in the
+  [community](https://github.com/knative/community) repo.
+  - Feedback is encouraged during a Technical Oversight Committee meeting before
+    any action is taken.
+- Once the TOC has approved of the removal, if the repo is not moving to another
+  actively maintained project:
+  - The repo description is edited to start with the phrase "[EOL]"
+  - All open issues and PRs are closed
+  - All external collaborates are removed
+  - All webhooks, apps, integrations, or services are removed
+  - GitHub pages are disabled
+  - The repo is marked as archived using GitHub's archive feature
+  - The removal is announced on the knative-dev mailing list
+
+This procedure maintains the complete record of issues, PRs, and other
+contributions. It leaves the repository read-only and makes it clear that the
+repository is retired and no longer maintained.
+
+## Additional / Experimental Repositories
+
+We want Knative working groups to be able to own code outside the core `knative`
+organization, which is kept intentionally small and has a high bar for entry.
+The `knative-sandbox` github organization was created for this purpose.
+Repositories in `knative-sandbox` are intended to give working groups more
+latitude to experiment with new ideas and to self-organize and manage
+contributions which may be important to the project but which do not need the
+level of governance of the core project.
+
+Note that `knative-sandbox` is **not** an incubation area for projects entering
+the core; in some cases, projects in the sandbox will remain there for the
+entire duration. Promotion from the sandbox into core will be handled on a
+case-by-case basis by joint decision of the Steering Committee and the Technical
+Oversight Committee.
+
+### Mechanics of working-group owned repositories in `knative-sandbox`
+
+#### Creation
+
+Note: prior art here from the
+[Kubernetes community](https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md).
+
+Working groups should be able to create repositories that meet the following
+bar:
```

Comment:
I'd like to defer the mechanics to a separate doc.

In terms of requiring TOC approval, I think there's a difference between needing the TOC to vote, and having someone on TOC execute some steps solo because they happen to have the keys. In the latter case, we could imagine replacing the TOC member with some automation or a designee who's granted access for the purpose of creating the repo. In the former case (TOC vote/approval), there's an expectation that the the TOC needs to actually understand what's being requested that the mechanism is secondary.